mantidproject/mantid
# Grid GUI controls inconsistent with plot generated from script.

**Description of work.**
When creating a plot from a script that was generated from a plot, the major grid lines visibility is now correctly reflected by the GUI controls. This includes the toolbar button and the checkbox inside the "Edit axis" dialog. The toolbar button will only be toggled on if every axis has major grid lines.

**To test:**
- Plot a spectrum from a workspace.
- Turn on major grid lines by clicking the toolbar button.
- Click the script button on the toolbar and select "Script to clipboard".
- Paste the script into the python editor and close the plot.
- Run the script and verify that the toolbar button for turning major grid lines on/off is correctly toggled.
- Double click an axis and check that the "Grid" checkbox is checked.

Additional testing (all covered by unit tests):
For each of the following, create a script from the plot, close the plot and run the script, and verify the GUI controls are as expected:
- Create a tiled plot and turn on grid lines.
- Create a tiled plot and turn on grid lines for only some of the plots (via the "Edit axis" dialog) - the toolbar button should be off when re-running the script.
- Create a spectrum plot and turn on the grid for only the x-axis (via the "Edit axis" dialog) - the toolbar button should be off when re-running the script.


Fixes #31279 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
